IDEAS/Buildings/Components/RectangularZoneTemplate.mo

```diff
@@ -33,27 +33,42 @@ model RectangularZoneTemplate
     annotation(Dialog(tab="Ceiling", group="Construction details"));
   parameter Boolean hasWinA = false
     "Modelling window for face A if true"
-    annotation(Dialog(tab="Face A", group="Window details"));
+    annotation(Dialog(tab="Face A", group="Window details", enable=not (bouTypA == IDEAS.Buildings.Components.Interfaces.BoundaryType.None)));
   parameter Boolean hasWinB = false
     "Modelling window for face B if true"
-    annotation(Dialog(tab="Face B", group="Window details"));
+    annotation(Dialog(tab="Face B", group="Window details", enable=not (bouTypB == IDEAS.Buildings.Components.Interfaces.BoundaryType.None)));
   parameter Boolean hasWinC = false
     "Modelling window for face C if true"
-    annotation(Dialog(tab="Face C", group="Window details"));
+    annotation(Dialog(tab="Face C", group="Window details", enable=not (bouTypC == IDEAS.Buildings.Components.Interfaces.BoundaryType.None)));
   parameter Boolean hasWinD = false
     "Modelling window for face D if true"
-    annotation(Dialog(tab="Face D", group="Window details"));
+    annotation(Dialog(tab="Face D", group="Window details", enable=not (bouTypD == IDEAS.Buildings.Components.Interfaces.BoundaryType.None)));
   parameter Boolean hasWinCei = false
     "Modelling window for ceiling if true"
-    annotation(Dialog(tab="Ceiling", group="Window details"));
+    annotation(Dialog(tab="Ceiling", group="Window details", enable=not (bouTypCei == IDEAS.Buildings.Components.Interfaces.BoundaryType.None)));
+  parameter Boolean hasIntZone = false "True if the zone contains an internal wall with both faces connected to itself"
+    annotation(Dialog(tab="Internal wall"));
   parameter Integer nSurfExt = 0
     "Number of additional connected external surfaces";
   parameter Modelica.SIunits.Angle aziA
     "Azimuth angle of face A";
   parameter Modelica.SIunits.Length l
-    "Horizontal length of faces A and C";
+    "Horizontal length of faces A and C. This parameter can be overwritten per surface";
   parameter Modelica.SIunits.Length w
-    "Horizontal length of face B and D";
+    "Horizontal length of faces B and D. This parameter can be overwritten per surface";
+  parameter Modelica.SIunits.Length lA = l
```

Comment:
These changes are from the other pull request. I will continue reviewing this file when the other pull request has been merged.